inspirehep/modules/records/api.py

```diff
@@ -586,9 +592,30 @@ def _append_current_control_number(control_number, base_name):
 
         return new_key
 
+    def _get_ref(self):
+        """Returns full url to this object (as in $ref)"""
+        pid = PersistentIdentifier.query.filter_by(object_uuid=self.id).one()
```

Comment:
Several remarks here:

1. the relation between PIDs and records is not one-to-one, as:
- in the future we will add non-INSPIRE PIDs like DOIs or arXiv IDs to the pidstore;
- already now, records can be merged, and then the PIDs of the two previous records both point to surviving one.
However, you don't need to ask the pidstore for the PID, as it's already in the record: `control_number` is the `pid_value` and the `$schema` encodes the `pid_type`.

2. **Not for this PR** because of the issue with merges, we need to be able to track citations across redirects. The current design is to have a task that periodically rewrites all redirected pids in the records: https://github.com/inspirehep/inspire-next/blob/dd0493e4f3fe9e6d1ddc17156b1ac3b604bbc8ac/inspirehep/modules/records/tasks.py#L48-L131 (this has never been run). Maybe it would be better to deal with it at search time instead, by returning all INSPIRE pids for a given record instead of only the canonical one?